impl/src/main/java/org/jboss/weld/event/EventMetadataImpl.java

```diff
@@ -59,6 +60,9 @@ private EventMetadataImpl(Type type, InjectionPoint injectionPoint, Set<Annotati
     public Set<Annotation> getQualifiers() {
         ImmutableSet.Builder<Annotation> builder = ImmutableSet.<Annotation> builder();
         builder.add(Any.Literal.INSTANCE);
+        if (qualifiers.isEmpty()) {
```

Comment:
The code below suggests that you should also look at `qualifierArray` (and that exactly one of `qualifiers` and `qualifierArray` is non-null).

Reply:
Point taken, will do